xbmc/utils/SystemInfo.cpp

```diff
@@ -867,6 +868,40 @@ bool CSysInfo::HasHW3DInterlaced()
   return false;
 }
 
+bool CSysInfo::HWSupportsStereo(const int mode)
+{
+#if defined(TARGET_ANDROID)
+  if (aml_present())
+    return aml_supports_stereo(mode);
+  else if (SysfsUtils::Has("/sys/class/graphics/fb0/3d_present"))  // AFTV
+    return true;
+#endif
+  return false;
+}
+
+void CSysInfo::HWSetStereoMode(const int mode, const int view)
+{
+#if defined(TARGET_ANDROID)
+  if (aml_present())
+    aml_set_stereo_mode(mode, view);
+  else if (SysfsUtils::Has("/sys/class/graphics/fb0/3d_present"))  // AFTV
+  {
+    switch(mode)
```

Comment:
I don't see RENDER_STEREO_MODE_INTERLACED supported here, but it's reported as supported a few lines above. Also the "interlaced" rendering on Android seems to be wrong. I tested both Kodi nightly and SPMC on the WETEK box and the interlacing (which should render each eye in alternating pixel rows/lines) seem to apply to the 720p framebuffer and thus got upscaled to 1080p which ofc broke the effect. Also the interlacing seemed to be done in columns not in rows, so it was interlaced ||| instead of =.